content/js/users.md

```diff
@@ -1,155 +1,295 @@
+  if (error.error.code === skygearError.InvalidCredentials ||
+      error.error.code === skygearError.ResourceNotFound ) {
+    // incorrect username or password
   } else {
-    console.log('user logged out or accessToken expired');
+    // other kinds of error
   }
-});
+})
+```
 
-handler.cancel(); // The callback is cancelable
+### Logging out
+
+Logging out the current user is simple using the `skygear.logout` method.
+
+Upon successful logout, the SDK will clear the current user and the access token
+from the local storage.
+
+``` javascript
+skygear.logout().then(() => {
+  console.log('logout successfully');
+}, (error) => {
+  console.error(error);
+});
 ```
 
 <a name="change-email-password"></a>
-## Change email/password
+## Changing email/password
+
+### Changing the email of a user
+
+To change a user's email, you can use the `skygear.saveUser` method by 
+providing the user ID and the new email.
+Every user can change his/her own email while
+only users with the admin role can change the emails of other users.
 
-### Change email
+Note: Changing the email of the current user will trigger the callback
+registered through `skygear.onUserChanged`, even if the new email is
+the same as the old one.
 
-By providing user id, current user can change his email.
-Only users with admin privileges can modify other users' email.
+To change the email of the current user:
 
 ``` javascript
 skygear.saveUser({
-  id: '<your-user-id>', // usually skygear.currentUser.id
-  email: '<your-new-email-address>',
+  id: skygear.currentUser.id,
+  email: 'new-email@example.com',
 }).then((user) => {
-  console.log('Email updated to:', user.email);
+  console.log(user); // updated user object
+  console.log('Email is changed to: ', user.email);
 }, (error) => {
   console.error(error);
 });
 ```
 
-### Change password
+### Changing the password of a user
 
-It's for current logged in user only.
+The currently logged-in user can change his/her own password.
+This can be done using the `skygear.changePassword` function.
+
+If the current password is incorrect, the SDK will return an
+`InvalidCredentials` error.
 
 ``` javascript
-skygear.changePassword(oldPassword, newPassword).then((user) => {
-  console.log('User now have new password', user);
+skygear.changePassword(currentPassword, newPassword).then((user) => {
+  console.log(user); // user object
+  console.log('Password has been changed');
 }, (error) => {
   console.error(error);
+  if (error.error.code === skygearError.InvalidCredentials) {
+    // the current password is incorrect
+  } else {
+    // other kinds of error
+  }
 });
 ```
 
-### Log out all other session (not yet implemented)
+Note: Changing the password of the current user will not trigger the callback
+registered through `skygear.onUserChanged`.
+
+[Not yet implemented] By setting `invalidateTokens` to `true`,
+you can invalidate the existing access token
+and get a new one so all other existing sessions become invalid.
```

Comment:
If we using JWT token store, all existing token will invalidate on changing password.

For other token, the invalidateTokens is not implemented.